### PR TITLE
Geometry_Engine: Added Query.FilterByNormal method

### LINKED ISSUE #2876: Geometry_Engine: Add  FilterByNormal method
#### Description:

Being able to filter a list of geometry objects by their normal would be really useful. For example, get every object which has an normal that is within a certain angle of a given vector.

### PR DESCRIPTION
**Pull request by tg359**
### Issues addressed by this PR

Closes #2876 

Added a method to filter a list of PlanarSurface objects by their surface normal, within a given tolerance.

### Test files

[Geometry_Engine-#2876-AddFilterByNormal.zip](https://github.com/BHoM/BHoM_Engine/files/9008323/Geometry_Engine-.2876-AddFilterByNormal.zip)

### Changelog

- `Query.FilterByNormal()` Query method added in the `Geometry_Engine` for lists of PlanarSurface objects

### CODE REVIEW

**Comment by tg359** [comment]
@BHoMBot check required

File: Geometry_Engine/Query/FilterByNormal.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        [Input("surfaces", "A list of PlanarSurfaces to filter by a given Normal vector.")]
+        [Input("vector", "The Vector to filter by.")]
+        [Input("tolerance", "Angle tolerance used in geometry processing, default set to BH.oM.Geometry.Tolerance.Angle.")]
+        [Output("matchingSurfaces", "A list of PlanarSurfaces where each surface Normal is within the tolerance angle of the given vector.")]
+        public static List<PlanarSurface> FilterByNormal(this List<PlanarSurface> surfaces, Vector vector, double tolerance = Tolerance.Angle)
+        {
+            List<PlanarSurface> matchingSurfaces = new List<PlanarSurface>();
+            foreach (PlanarSurface srf in surfaces)
+            {
+                if (Math.Abs(Query.Angle(vector, srf.Normal())) < tolerance)
```
Comment (pawelbaran) [inline Geometry_Engine/Query/FilterByNormal.cs:48]: You could use https://github.com/BHoM/BHoM_Engine/blob/main/Geometry_Engine/Query/IsParallel.cs here - additionally, possibly worth considering adding a `bool` method input to specify if the opposite vectors should be treated as equal?

Comment (tg359) [reply Geometry_Engine/Query/FilterByNormal.cs]: I did what you suggested, so the method's a little more flexible now. 

I'd like to have it a bit more generalised (applied to lists of like-objects), but as it's being driven by a pretty recurrent need of mine I'm happy with just the `PlanarSurface` version for now.

**Comment by tg359** [comment]
@BHoMBot check required

**Review by pawelbaran — APPROVED** [review]
Thank you @tg359 for including my suggestions in the code. Now I am happy to approve it - it does what it is meant to do.

**Comment by pawelbaran** [comment]
@BHoMBot check required

**Comment by pawelbaran** [comment]
@BHoMBot check versioning
@BHoMBot check copyright-compliance
@BHoMBot check dataset-compliance
@BHoMBot check ready-to-merge

**Pull request merged**